scripts/find_localised_strings.py

```diff
@@ -74,69 +70,52 @@ def extract_comments(call: ast.Call, lines: list[str], file: pathlib.Path) -> Op
     :param file: The path to the file this call node came from
     :return: The first comment that matches the rules, or None
     """
-    out: Optional[str] = None
-    above = call.lineno - 2
-    current = call.lineno - 1
-
-    above_line = lines[above].strip() if len(lines) >= above else None
-    above_comment: Optional[str] = None
-    current_line = lines[current].strip()
-    current_comment: Optional[str] = None
-
-    bad_comment: Optional[str] = None
-    if above_line is not None:
-        match = COMMENT_OWN_LINE_RE.match(above_line)
-        if match:
-            above_comment = match.group(1).strip()
-            if not above_comment.startswith('# LANG:'):
-                bad_comment = f'Unknown comment for {file}:{call.lineno} {above_line}'
-                above_comment = None
-
-            else:
-                above_comment = above_comment.replace('# LANG:', '').strip()
+    above_line_number = call.lineno - 2
+    current_line_number = call.lineno - 1
 
-    if current_line is not None:
-        match = COMMENT_SAME_LINE_RE.match(current_line)
-        if match:
-            current_comment = match.group(1).strip()
-            if not current_comment.startswith('# LANG:'):
-                bad_comment = f'Unknown comment for {file}:{call.lineno} {current_line}'
-                current_comment = None
-
-            else:
-                current_comment = current_comment.replace('# LANG:', '').strip()
+    def extract_lang_comment(line: str) -> str | None:
+        """
+        Extract a language comment from a given line.
 
-    if current_comment is not None:
-        out = current_comment
+        :param line: The line to extract the language comment from.
+        :return: The extracted language comment, or None if no valid comment is found.
+        """
+        match = COMMENT_OWN_LINE_RE.match(line)
+        if match and match.group(1).startswith('# LANG:'):
+            return match.group(1).replace('# LANG:', '').strip()
+        return None
 
-    elif above_comment is not None:
-        out = above_comment
+    above_comment = extract_lang_comment(lines[above_line_number]) if len(lines) >= above_line_number else None
+    current_comment = extract_lang_comment(lines[current_line_number])
 
-    elif bad_comment is not None:
-        print(bad_comment, file=sys.stderr)
+    if current_comment is None:
+        current_comment = above_comment
 
-    if out is None:
-        print(f'No comment for {file}:{call.lineno} {current_line}', file=sys.stderr)
+    if current_comment is None:
+        print(f'No comment for {file}:{call.lineno} {lines[current_line_number]}', file=sys.stderr)
+        return None
 
-    return out
+    return current_comment
```

Comment:
The old code checked if a comment didn't match the expected syntax and printed a message for that, the new code does not. Both codes do mention those lines as having no comment as well.